### PR TITLE
Add NVENC AV1 support

### PR DESCRIPTION
**Pull request by jp9000**
### Description
Adds NVENC AV1 encoding support

### Motivation and Context
AV1 is super sweet and we need to encourage the industry to move forward on newer, better codecs

### How Has This Been Tested?
On a test machine with applicable hardware

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.

### CODE REVIEW

File: UI/window-basic-main-profiles.cpp [hunk 1]
```diff
@@ -862,6 +862,12 @@ void OBSBasic::CheckForSimpleModeX264Fallback()
 				name = SIMPLE_ENCODER_X264;
 				return false;
 			}
+		} else if (strcmp(name, SIMPLE_ENCODER_NVENC_AV1) == 0) {
+			if (!nve_supported) {
+				changed = true;
+				name = SIMPLE_ENCODER_X264;
```
Comment (Conan-Kudo) [inline UI/window-basic-main-profiles.cpp:868]: This should fallback to rav1e, svt, or aom here, shouldn't it? Otherwise this makes no sense.

Comment (jp9000) [reply UI/window-basic-main-profiles.cpp:868]: You know, I think I'd rather make it just explicitly fail than fall back to a software AV1 encoder. Software AV1 encoders aren't really usable for most people due to their core requirements

Comment (Conan-Kudo) [reply UI/window-basic-main-profiles.cpp:868]: Are we sure about that? Realtime AOM modes seemed to work reasonably okay on my PCs from 2014 (4th generation Intel) and 2016 (6th generation Intel).

Comment (jp9000) [reply UI/window-basic-main-profiles.cpp:868]: If it did, it probably didn't compress very well.

Comment (Conan-Kudo) [reply UI/window-basic-main-profiles.cpp:868]: But does that matter when you're streaming to the other end? Realtime encoding trades off compression for quality and speed.

Comment (jp9000) [reply UI/window-basic-main-profiles.cpp:868]: I suppose I'm fine with falling back to AOM or something.

Comment (Conan-Kudo) [reply UI/window-basic-main-profiles.cpp:868]: That would be great, I know I would certainly appreciate it!

Comment (RytoEX) [reply UI/window-basic-main-profiles.cpp:868]: I don't think falling back from HW encoder to SW encoder is a good idea. Just select the SW encoder if you want to use that. Error out with a helpful message if we can't use the HW encoder.

Comment (Fenrirthviti) [reply UI/window-basic-main-profiles.cpp:868]: Also confirming falling back from a hardware to a software encoder is a bad idea. We don't fall back to x264 for any of the other hardware h264 encoders, and I don't see a reason why we should start doing that for those or AV1.

File: UI/window-basic-main-outputs.cpp [hunk 2]
```diff
@@ -290,9 +290,7 @@ struct SimpleOutput : BasicOutputHandler {
 	void UpdateRecordingSettings_x264_crf(int crf);
 	void UpdateRecordingSettings_qsv11(int crf);
 	void UpdateRecordingSettings_nvenc(int cqp);
-#ifdef ENABLE_HEVC
```
Comment (Conan-Kudo) [inline UI/window-basic-main-outputs.cpp:293]: If you're going to remove these guards, can you do it in a separate commit?

File: UI/window-basic-main-profiles.cpp (hunk 1 above)
Comment (Conan-Kudo) [reply UI/window-basic-main-profiles.cpp:868]: I'm fine with no fallback at all too. Though I'd love to eventually see an AV1 alternative to the x264 plugin.

Comment (jp9000) [reply UI/window-basic-main-profiles.cpp:868]: We already have AOM and SVT.

File: UI/window-basic-main-outputs.cpp (hunk 2 above)
Comment (tytan652) [reply UI/window-basic-main-outputs.cpp:293]: Those guards are removed because `UpdateRecordingSettings_nvenc_hevc` is used for AV1.

Maybe `UpdateRecordingSettings_nvenc_hevc` needs to be renamed since it is no longer HEVC only.

Comment (Conan-Kudo) [reply UI/window-basic-main-outputs.cpp:293]: Yeah, that would be a good idea.

File: UI/window-basic-main-profiles.cpp (hunk 1 above)
Comment (jp9000) [reply UI/window-basic-main-profiles.cpp:868]: This code doesn't have anything to do with silently falling back. This code is for when you start up obs and the encoder no longer exists with the current config anymore (i.e. switched parts, copied config from another computer).

File: UI/window-basic-main-outputs.cpp (hunk 2 above)
Comment (jp9000) [reply UI/window-basic-main-outputs.cpp:293]: I changed the name of the function to be more clear.
